src/main/js/dygraph-light/main/LabelMaker.ts

```diff
@@ -200,25 +199,26 @@ class Metadata {
 	}
 }
 
-const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabels: string[], lblIdx: number, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
-	if (linking === 'concatenate'){
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: getLabel(object.tableFormat, colName)) + postfix;
+const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabel: string, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
+	if (requestedLabel) return requestedLabel + postfix;
 
-	} else {
-		const filename = object.filename;
+	if (linking === 'concatenate'){
+		return getLabel(object.tableFormat, colName) + postfix;
+	} 
+	const filename = object.filename;
 
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: filename.slice(0, filename.lastIndexOf('.'))) + postfix;
-	}
+	return filename.slice(0, filename.lastIndexOf('.')) + postfix;
 };
 
-const getLegendLabelsFromParams = (params: UrlSearchParams, name: string) => {
-	return params.has(name)
-		? params.get(name).split(',') as string[]
-		: [];
+const getLegendLabelFromParams = (params: UrlSearchParams, name: string, objId: string) => {
+	const labels: string[] = params.has(name) ? params.get(name).split(',') : [];
+	const ids: string[] = params.has('objId') ? params.get('objId').split(',') : [];
+	
+	if (labels.length === 0 || ids.length === 0) return '';
+	
+	const idx = ids.findIndex(id => objId.includes(id));
+
+	return (idx !== -1 && labels[idx]) ? labels[idx] : '';
```

Comment:
I think this is equivalent?
```suggestion
	return labels[idx] || '';
```